mindmeld/components/entity_resolver.py

```diff
@@ -151,36 +153,78 @@ def create_resolver(cls, app_path, entity_type, config=None, resource_loader=Non
 
         return ENTITY_RESOLVER_MODEL_MAPPINGS.get(resolver_type)(
             app_path,
-            entity_type=entity_type,
+            entity_type,
             config=er_config,
             resource_loader=resource_loader,
             **kwargs)
 
 
-class BaseEntityResolver(ABC):
+class EntityResolver:
+    """
+    Class for backwards compatability
+
+    deprecated usage
+        >>> entity_resolver = EntityResolver(
+                app_path, resource_loader, entity_type
+            )
+
+    new usage
+        >>> entity_resolver = EntityResolverFactory.create_resolver(
+                app_path, entity_type
+            )
+        # or ...
+        >>> entity_resolver = EntityResolverFactory.create_resolver(
+                app_path, entity_type, resource_loader=resource_loader
+            )
+    """
+
+    def __new__(cls, app_path, resource_loader, entity_type, **kwargs):
+        msg = "Entity Resolver should now be loaded using EntityResolverFactory. " \
+              "See https://www.mindmeld.com/docs/userguide/entity_resolver.html for more details."
+        warnings.warn(msg, DeprecationWarning)
+        return EntityResolverFactory.create_resolver(
+            app_path, entity_type, resource_loader=resource_loader, **kwargs
+        )
+
+
+class BaseEntityResolver(ABC):  # pylint: disable=too-many-instance-attributes
     """
     Base class for Entity Resolvers
     """
 
     def __init__(self, app_path, entity_type, resource_loader=None, **_kwargs):
-        """Initializes an entity resolver"""
+        """Initializes an entity resolver
+
+        Args:
+            app_path (str): The application path.
+            entity_type (str): The entity type associated with this entity resolver.
+            resource_loader (ResourceLoader, Optional): An object which can load resources for the
+                resolver.
+        """
         self.app_path = app_path
         self.type = entity_type
         self._resource_loader = (
             resource_loader or ResourceLoader.create_resource_loader(app_path=self.app_path)
         )
 
-        self._processed_entity_map = None
         self._is_system_entity = Entity.is_system_entity(self.type)
         self._no_trainable_canonical_entity_map = False
-        self.dirty = False  # bool, True if exists any unsaved generated data that can be saved
-        self.ready = False  # bool, True if the model is fit by calling .fit()
+        self.dirty = False  # bool, True if exists any unsaved data/model that can be saved
+        self.ready = False  # bool, True if the model is already fitted or loaded
+        self.hash = None
+
+        # resources for creating and maintaining processed entity map
+        self._aug_lower_case = None
+        self._aug_title_case = None
+        self._aug_normalized = None
+        self._normalize_aliases = None
```

Comment:
Aren't these specific to non-ES resolvers?

Reply:
They are! Moving them back to the non-ES resolvers from the base resolver class.